string sender
string destination
string type
